api/v1beta1/types.go

```diff
@@ -100,7 +100,7 @@ type NetworkSpec struct {
 
 	// APIServerLB is the configuration for the control-plane load balancer.
 	// +optional
-	APIServerLB LoadBalancerSpec `json:"apiServerLB,omitempty"`
+	APIServerLB *LoadBalancerSpec `json:"apiServerLB,omitempty"`
```

Comment:
Instead of using a pointer, what about moving the additional field you introduced (`ControlPlaneEnabled`) here?

With such approach we could introduce the feature smoothly with no breaking changes, and maintain the current state of the pointers.

Reply:
The "problem" is that der is a mutation inside the webhook and if this is no pointer then some fields will be filled which makes problems later. To me the question is `omitempty` means it can be null so the pointer also makes sense to me.